### PR TITLE
Refactor CMake support for libraries (Fixes #261)

### LINKED ISSUE #261: Issues when linking .so libs on Linux
CMake does not appear to support linking shared library objects for the MKL linux installation.

### PR DESCRIPTION
**Pull request by guacke**
This PR will gather CMake refactors aimed at making the build generation procedure more robust.

### CODE REVIEW

**Comment by guacke** [comment]
I have reverted the last commit, because I missed the Spectra dependency.

**Comment by guacke** [comment]
Please take the additional details in the README into account before attempting to use this branch.

File: cmake/Find/FindCint.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#[=======================================================================[.rst:
+FindCint
+--------
+
+Find the Cint library (https://github.com/sunqm/libcint) on the system.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module will set the following variables in your project:
```
Comment (lelemmen) [inline cmake/Find/FindCint.cmake]: Could you also add to the documentation that a target for the Libcint library is made?

File: cmake/Find/FindCint.cmake [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+find_package_handle_standard_args(Cint REQUIRED_VARS
+        Cint_INCLUDE_DIR Cint_LIBRARY)
+
+if(Cint_FOUND AND NOT TARGET Cint::Cint)
+    add_library(Cint::Cint UNKNOWN IMPORTED)
+    set_target_properties(Cint::Cint PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES ${Cint_INCLUDE_DIR}
+            IMPORTED_LOCATION ${Cint_LIBRARY}
+            )
+endif()
```
Comment (lelemmen) [inline cmake/Find/FindCint.cmake]: add newline at eof

File: cmake/Find/FindCint.cmake [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  Define the root directory of the Cint installation.
+#]=======================================================================]
+
+find_path(Cint_INCLUDE_DIR cint.h HINTS ${Cint_ROOT_DIR}/include)
+find_library(Cint_LIBRARY cint HINTS ${Cint_ROOT_DIR}/lib)
+mark_as_advanced(Cint_INCLUDE_DIR Cint_LIBRARY)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Cint REQUIRED_VARS
+        Cint_INCLUDE_DIR Cint_LIBRARY)
```
Comment (lelemmen) [inline cmake/Find/FindCint.cmake]:
```suggestion
        Cint_INCLUDE_DIR Cint_LIBRARY)  # sets Cint_FOUND
```

File: cmake/Find/FindInt2.cmake [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#[=======================================================================[.rst:
+FindInt2
+--------
+
+Find the Int2 library (https://github.com/evaleev/libint) on the system.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module will set the following variables in your project:
```
Comment (lelemmen) [inline cmake/Find/FindInt2.cmake]: idem libcint

File: cmake/Find/FindInt2.cmake [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+``Int2_ROOT_DIR``
+  Define the root directory of the Int2 installation.
+#]=======================================================================]
+find_path(Int2_INCLUDE_DIR libint2.hpp HINTS ${Int2_ROOT_DIR}/include)
+find_library(Int2_LIBRARY int2 HINTS ${Int2_ROOT_DIR}/lib)
+mark_as_advanced(Int2_INCLUDE_DIR Int2_LIBRARY)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Int2 REQUIRED_VARS
+        Int2_INCLUDE_DIR Int2_LIBRARY)
```
Comment (lelemmen) [inline cmake/Find/FindInt2.cmake]:
```suggestion
        Int2_INCLUDE_DIR Int2_LIBRARY)  # sets Int2_FOUND
```

File: cmake/Find/FindInt2.cmake [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+if(Int2_FOUND AND NOT TARGET Int2::Int2)
+    add_library(Int2::Int2 UNKNOWN IMPORTED)
+    set_target_properties(Int2::Int2 PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES ${Int2_INCLUDE_DIR}
+            IMPORTED_LOCATION ${Int2_LIBRARY}
+            )
+endif()
```
Comment (lelemmen) [inline cmake/Find/FindInt2.cmake]: add newline at eof

**Comment by guacke** [comment]
I agree that the new (lib)Cint and (lib)Int2 names sound artifical, but the use of 'Lib' by the developers of these packages is even stranger. E.g., we do not link to 'LibEigen' or 'LibMKL'. Furthermore, when linking, we use -lcint or -lint2 and not -llibcint or -llibint2. @tmhuysen What do you think?

**Comment by guacke** [comment]
These commits will are also contained in PR #279.

**Pull request closed without merging**